docs/docs/functions/macros.md

```diff
@@ -39,20 +57,31 @@ Defines new blocks that can be inserted into an existing chain.
 
 Defines a new non-looped chain.
 
-`(defchain my-chain)` is actually a shorthand for the more verbose non-looped chain definition.
 === "Code"
 
     ```clojure linenums="1"
     ;; defchain
+    (defchain my-chain
+        ;; blocks here
+    )
+    ```
+
+`(defchain <chain-name>)` is actually a shorthand for the more verbose non-looped chain definition: `def <chain-name>` + `Chain "chain-name"`.
+=== "Code"
+
+    ```clojure linenums="1"
+    ;; def + Chain
     (def my-chain
     (Chain "my-chain"
         ;; blocks here
     ))
     ```
 
-For a chain to be executed, it must first be scheduled on a `node` and then that that node needs to run.
+A chain is a stateful function.
```

Comment:
this is a good statement but not in the right place and maybe needs to be continued?

Reply:
I think we will later have a link to a more in-depth article about chain and blocks here.
We could already set that link and point to a "TODO" or "coming soon" placeholder page.

Reply:
I agree we should elaborate on this. IMHO this should go in both the 'Chainblocks Architecture' document as well as the 'Chainblocks Primer' document. Both should talk about this in slightly different contexts (the former in terms of blockchain elements, the latter in terms of programming).
I have added the expected links to this page. The links are currently broken but I have made a note to modify these links accordingly when I put content in the chainblocks architecture and primer docs (next in my queue).

Reply:
Added TODO pages for :
> Chainblocks architecture guide with [this commit](https://github.com/fragcolor-xyz/chainblocks/pull/238/commits/2db79b473ee45e88eeebd10a624e779c4616b6f1)
> Chainblocks primer doc with [this commit](https://github.com/fragcolor-xyz/chainblocks/pull/238/commits/c595e1fb6692c4d86b3e92096a39648a6307d516)